### PR TITLE
week-4-assessment-completed

### PR DESCRIPTION
**Pull request by Jk0ng**
# Submitting Week 4 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Career Readiness
- [x] Coding Challenges
- [x] Interview Questions
- [x] Ruby Challenges

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

**Comment by Jk0ng** [comment]
I used function(){return 0.5 - Math.random()} to shuffle the array for the first question in JavaScript coding challenges because I was looking up built-in methods to shuffle an array but found this function(){return 0.5 - Math.random()}. I tried to look into this function but still not really sure I have a good grasp on it. So, some explanation to how setting the value to -0.5 to 0.5 shuffle the array will be appreciated!

File: code-challenges.test.js [hunk 1]
```diff
@@ -20,19 +20,58 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
 // Pseudo code:
+// input: an array 
+// output: the array with the first item removed and shuffled. 
+// create a function that takes in an array
+// use .slice method to return  all the items after the first element [1] from the OG array in the form of an array 
+//  then use .sort() function to sort the new returned array and set the comparison function  by using 0.5 - Math.random, and this give a range from -0.5 to 0.5. The sort function will sort the array in an ascending order, but the random values from -0.5 to 0.5 randomize the order.  
+
+const function1 = (array) => {
+    return array.slice(1).sort(function(){return 0.5 - Math.random()})
+}
```
Comment (triciasykes) [inline code-challenges.test.js:44]: I LOVE THIS!  great code, excellent description of how to use the sort method to randomize.  Well done, Joseph!

File: code-challenges.test.js [hunk 2]
```diff
@@ -20,19 +20,58 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+// // create a function called voteCount that takes in an obeject. 
+// // use obejct.value to print out all the values in an array. the result will look like this [# of upvotes, # of downvotes]
+// // assign this^^ to a newArray
+// // then add the two elemnets by using bracket notation; newArray[0] - newArray[1] 
+
+const voteCount = (object) => {
+    const newArray = Object.values(object)
+    return newArray[0] - newArray[1]
+}
+
```
Comment (triciasykes) [inline code-challenges.test.js:76]: Interesting approach.  I like that you're exploring different ways to do things!

File: ruby-challenges.rb [hunk 3]
```diff
@@ -11,6 +11,24 @@
...
+def even_or_odd (integer)
+    unless integer.even? 
+        "#{integer} is odd"
+    else 
+        "#{integer} is even"
+    end
+end
+p even_or_odd (reposts1)
+p even_or_odd (reposts2)
+p even_or_odd (reposts3)
```
Comment (triciasykes) [inline ruby-challenges.rb:30]: You can do it with the `unless`, but more conventional to use `if`.  Either way it works!  Again, thinking outside of the box. :)

File: ruby-challenges.rb [hunk 4]
```diff
@@ -33,4 +64,22 @@
...
+    if string.downcase == string.downcase.reverse
+        "#{string} is a palindrome"
+    else 
+        "#{string} is not a palindrome"
+    end
+end
+
+p palindrome(palindrome_test_case1)
+p palindrome(palindrome_test_case2)
+p palindrome(palindrome_test_case3)
```
Comment (triciasykes) [inline ruby-challenges.rb:85]: BOOM! 💥

**Review by triciasykes — APPROVED** [review]
Joseph Kong - you crushed this assessment.  Fantastic logic and pseudocode.  I'm really impressed.  Your interview answers show you grasp the concepts.  I think the thing that would help you most is to start with the simple solution.  I think you made a couple of the problems a bit more complicated (ie more steps) than necessary.  But we all have to grow. :). AWESOME job!

**Pull request merged**